### PR TITLE
Migrate BraveViewHostMsg_* legacy IPC messages to mojo #15546

### PR DESCRIPTION
**Pull request by mariospr**
These are the only messages present in common/render_messages.h, and
also the reason why we define a "random" value for IPC_MESSAGE_START
in that file, so let's migrate them to mojo and fix two problems at
once, since legacy IPC is deprecated anyway.

Resolves https://github.com/brave/brave-browser/issues/15546

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Open the browser and go to "Settings" > "Shields"
2. Enable the "Block Scripts" option, that will block scripts globally on any URL
3. Navigate to https://coveryourtracks.eff.org and open the Brave Shields panel on the address bar
4. A non-zero number should be shown under the "Scripts blocked" section

### CODE REVIEW

**Review by bridiver — APPROVED** [review]
lgtm assuming there are tests covering this

**Comment by mariospr** [comment]
> lgtm assuming there are tests covering this

@bridiver I've checked manually and this seems to be working fine, but I'm afraid **none of the current tests cover this change** so tomorrow I'll try to add one if I can, or specify testing steps otherwise.

Btw, I've also added another commit on top to remove the `OnFingerprintBlocked()` method as per the internal discussion with the security team. It seems the right call to do considering that such method is never invoked, but in any case it would be good if we could get confirmation from @simonhong as the person who dropped the -now removed- former caller.

**Comment by mariospr** [comment]
@bridiver Finally managed to write a browser test to check that this migration actually worked as expected, see the third commit "Add a new browser test suite for BraveShieldsWebContentsObserverTest" in this PR.

Since you've already approved this assuming there was test coverage, I'll be merging this PR as soon as the CI bots confirm that the new test runs and passes fine on all platforms, so that we can also unblock https://github.com/brave/brave-core/pull/8678 ASAP, hope that's ok.

**Pull request merged**
